services/task-svc/internal/repositories/patient_repository.go

```diff
@@ -130,3 +130,13 @@ func (r *PatientRepository) DeletePatient(patientID uuid.UUID) (*models.Patient,
 	}
 	return patient, nil
 }
+
+func (r *PatientRepository) ReactivatePatient(patientID uuid.UUID) (*models.Patient, error) {
+	patient := &models.Patient{ID: patientID}
+	query := r.db.Unscoped().Model(patient).Update("is_discharged", 0)
```

Comment:
is `Unscoped` needed here? from what I can gather, it only relates to gorm's soft-delete mechanic

Reply:
Yes, and as we look for the soft deleted patients to reactivate, we need the `Unscope`